.github/workflows/github-publish.yaml

```diff
@@ -61,14 +61,77 @@ jobs:
           SIGNING_KEY: ${{ secrets.SIGNING_KEY }}
           SIGNING_PASSWORD: ${{ secrets.SIGNING_PASSWORD }}
 
-      - name: Publish Images
+      - name: Publish Operator Images
         uses: gradle/gradle-build-action@v2
         if: ${{ inputs.module == 'operator' }}
         with:
-          arguments: pushCRD pushDocker pushHelm -PdockerRegistry=${{ steps.login-ecr-public.outputs.registry }}/j8q9y0n6 -PhelmRegistry=${{ steps.login-ecr-public.outputs.registry }}/j8q9y0n6
+          arguments: operator:pushCRD operator:pushDocker operator:pushHelm -PdockerRegistry=${{ steps.login-ecr-public.outputs.registry }}/j8q9y0n6 -PhelmRegistry=${{ steps.login-ecr-public.outputs.registry }}/j8q9y0n6
         env:
           ECR_REGISTRY: ${{ steps.login-ecr-public.outputs.registry }}
 
+      - name: Publish Example Images
+        uses: gradle/gradle-build-action@v2
+        if: ${{ inputs.module == 'kafka-client' }}
+        with:
+          arguments: kafka-client-examples:simple-example:pushDocker -PdockerRegistry=${{ steps.login-ecr-public.outputs.registry }}/j8q9y0n6 -PhelmRegistry=${{ steps.login-ecr-public.outputs.registry }}/j8q9y0n6
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr-public.outputs.registry }}
+
+      - uses: actions/checkout@v3
+        with:
+          repository: responsivedev/sindri
+          path: sindri
+          ref: refs/heads/master
+          token: ${{ secrets.TOOLS_GHA_TOKEN }}
+
+      - name: Install Kind
+        uses: helm/kind-action@v1.5.0
+        with:
+            install_only: true
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: arn:aws:iam::292505934682:role/github-responsive-pub-main
+          role-session-name: github-publish-artifacts
+          aws-region: us-west-2
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Set Operator Version
+        run: |
+          echo "OPERATOR_VERSION=$(./gradlew operator:cV | grep Project.version | sed 's/Project version: //')" >> $GITHUB_ENV
+
+      - name: Set Kafka Client Version
+        run: |
+          echo "KAFKA_CLIENTS_VERSION=$(./gradlew kafka-client:cV  | grep Project.version | sed 's/Project version: //')" >> $GITHUB_ENV
+
+      - name: Run Smoke Test (Operator)
```

Comment:
if either of these fail, we probably want to "unpublish" the images, right? or is there a way for us not to publish them to the public location at all until after we've tested them?

Reply:
yeah I was thinking about this and landed on either:
- we can just cut a new release
- we can setup a release patching process.
I think either approach is fine for where we're at now.

I swear I did consider running these before publishing but now I can't remember why I didn't implement it that way. Something didn't work when I tried. Let me think about it. I agree that would be preferable.

Reply:
I'll try to make it work. I'm just going to merge this first. And then do a follow-up with that change.